fixture-monkey/src/main/java/com/navercorp/fixturemonkey/resolver/ArbitraryTree.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Fixture Monkey
+ *
+ * Copyright (c) 2021-present NAVER Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.navercorp.fixturemonkey.resolver;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+import org.apiguardian.api.API;
+import org.apiguardian.api.API.Status;
+
+import net.jqwik.api.Arbitraries;
+import net.jqwik.api.Arbitrary;
+
+import com.navercorp.fixturemonkey.api.generator.ArbitraryGeneratorContext;
+import com.navercorp.fixturemonkey.api.generator.ArbitraryProperty;
+import com.navercorp.fixturemonkey.api.option.GenerateOptions;
+
+@API(since = "0.4.0", status = Status.EXPERIMENTAL)
+final class ArbitraryTree {
+	private final ArbitraryNode rootNode;
+	private final GenerateOptions generateOptions;
+
+	ArbitraryTree(
+		ArbitraryNode rootNode,
+		GenerateOptions generateOptions
+	) {
+		this.rootNode = rootNode;
+		this.generateOptions = generateOptions;
+	}
+
+	Arbitrary<?> generate() {
+		return this.generateOptions.getArbitraryGenerator(rootNode.getProperty())
```

Comment:
아래처럼 generateContext 메소드 호출을 한번 분리하면 좋을거 같습니다.
(메소드 체인이 많고, 그 안에 nested 로 메소드 호출이 있으니까 읽는데 에너지가 많이 사용되네요)


```suggestion
        ArbitraryGeneratorContext context = this.getGenerateContext(rootNode, null);
		return this.generateOptions.getArbitraryGenerator(rootNode.getProperty())
		    .generate(context);
```